modules/puppet_proxy/puppet_api.rb

```diff
@@ -7,14 +7,6 @@ class Proxy::Puppet::Api < ::Sinatra::Base
 
   inject_attr :class_retriever_impl, :class_retriever
   inject_attr :environment_retriever_impl, :environment_retriever
-  inject_attr :puppet_runner_impl, :puppet_runner
-
-  post "/run" do
```

Comment:
Thinking out loud: is it a good idea to maintain the endpoint but return an explicit HTTP 501 Not Implemented?

Reply:
makes sense, how do we make sure we drop this in next major release? do we have redmine issue for that? do we have deprecation messages similar to core?

Reply:
I think we need some proper maintenance on the Smart Proxy project. There are many things we deprecated, but never removed. ee1dcd16bdba277294cb9c744b667f911e1ea7c7 for example stated the methods would be removed in 1.24 but haven't to this day. 5427599db2fbf258d7513260e852ad44ee41ef2c (part of 1.23) removed methods deprecated in 1.14 / 1.16. Then we have the `LegacyModuleLoader` which is still used by 2 modules without a clear plan to remove them.

In this case it's an external API. Keeping it around makes it easier to debug. Its usefulness is gone when there's no real chance a Foreman with Puppetrun functionality has a Smart Proxy. While in many cases we say Foreman should be newer than the Proxy, in reality I still think Dmitri was right that just Foreman <-> Foreman Proxy it's much easier to say that Foreman Proxy should be upgraded first. It's just that with Pulp that may not be true.

In short: I'd keep this around for at least 2 releases but probably 4 just to be safe. Or wait for the next major version.